include/wx/dnd.h

```diff
@@ -44,6 +44,15 @@ enum wxDragResult
     wxDragCancel    // the operation was cancelled by user (not an error)
 };
 
+// Flags for wxDataViewEvent::GetDragFlags() with wxEVT_DATAVIEW_DROP
+// Note: 0x05 & 0x06 are valid combinations!
+enum wxDropFlags
```

Comment:
Naming-wise it's not great to use `wxDropFlags` as return value of `GetDragFlags()`, "drag" and "drop" are different verbs. It looks like `GetDragFlags()` is actually not used at all right now, which is, I guess, why you've decided to reuse it, but I'd still rather add `GetDropFlags()` if we want to return these flags.

But more importantly, I'd still like to somehow reuse `GetProposedDropIndex()` for this because otherwise you simply can't write portable code working correctly. I.e. you can either use the proposed drop index and it would work in the generic and Cocoa versions (i.e. under MSW and Mac) or use the drop flags and then it would work with wxGTK, but there is no way to make it work with both which is clearly not ideal.

Even if we can't represent all the possibilities using the proposed drop index, we should at least at least fill it when we can. I.e. it would be fine to provide some extra information in wxGTK which is not available under the other platforms, but we need to have at least some functionality common to all implementations.

Reply:
My intention is to at least add the new flags to the generic dataview, but this looks like it will take a while to figure out and unfortunately other things have been making calls on my time. Might be able to do some trickery with GTK's `map_expanded_rows` in order to implement `GetProposedDropIndex` with relatively few lines of code, but not sure how closely it will emulate how generic does things.

Probably should tag this as `WIP` for the time being.

Reply:
I think there is an inevitable API break here.
In cases where the item is dropped between two rows, wxGTK will select the lower row as what ends up returned by event.GetItem() whereas wxMSW (i.e. generic) will select the parent of the rows. Which one should be adopted as the universal convention?

Reply:
> I think there is an inevitable API break here.

This is not necessarily catastrophic (although best avoided, of course), the only "hard" requirement is to have at least a useful subset of the API actually working the same on all platforms, but I don't even think we need to break the API here.

> In cases where the item is dropped between two rows, wxGTK will select the lower row as what ends up returned by event.GetItem() whereas wxMSW (i.e. generic) will select the parent of the rows. Which one should be adopted as the universal convention?

The latter behaviour seems more useful to me and this is also how the native macOS version behaves, so I'd strongly prefer to keep it like this and make wxGTK behave in the same way. Would it be possible to do it like this please? TIA!

Reply:
I'll see what i can put together.

Reply:
Did some quick testing with `samples/dataview` which looks like it was made for wxGeneric's reporting convention.